src/lib/data/near.js

```diff
@@ -337,22 +337,43 @@ async function _initNear({
 }
 
 export const useInitNear = singletonHook({}, () => {
-  const [nearPromise, setNearPromise] = useState(null);
-
-  return {
-    nearPromise,
-    initNear: useMemo(() => (args) => setNearPromise(_initNear(args)), []),
-  };
+    const [nearPromise, setNearPromise] = useState(null);
+
+    return {
+        nearPromise,
+        initNear: useMemo(
+            () => (args) => {
+                const defaultNetworkId = args.config?.networkId || args.networkId;
+                const testnetArgs = defaultNetworkId === 'testnet' ? args : { ...args, networkId: 'testnet', config: undefined, keyStore: undefined, selector: undefined };
+                const mainnetArgs = defaultNetworkId === 'mainnet' ? args : { ...args, networkId: 'mainnet', config: undefined, keyStore: undefined, selector: undefined };
+                return setNearPromise(Promise.all([testnetArgs, mainnetArgs].map(_initNear)).then((nears) => nears.map((n) => ({ ...n, default: n.config.networkId === defaultNetworkId }))));
```

Comment:
It might be an issue, that we'll get a `networkId` that is not `mainnet` or `testnet`. In this case the VM will not use `args` at all. E.g. for a `localnet`.

Reply:
Passed `args` through if `networkId` is something else. `default` should be `true` in this case 👍